src/graph/FromArrayPointsNode.cpp

```diff
@@ -0,0 +1,45 @@
+// Copyright 2023 Robotec.AI
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <graph/NodesCore.hpp>
+
+void FromArrayPointsNode::setParameters(const void* points, size_t pointCount, const std::vector<rgl_field_t>& fields)
+{
+	// TODO(msz-rai): Optimize memory allocation. Do not clear all fields.
```

Comment:
I don't mind clearing all fields. It's a rare operation, optimizing here is premature IMO. CC @PiotrMrozikRobotec

Reply:
I could imagine a case where we would like to test the sequence of point clouds. With the increasing number of tests, it would be nice to reduce execution time.